### DISCUSSION TITLE
Save and Retrieve Blazor Diagram

### DISCUSSION THREAD

**Discussion by ecadversario** [Blazor-Diagrams/Blazor.Diagrams#391] [General]
Hi guys, anybody who knows how can I save and retrieve Blazor.Diagrams?

**Comment by kobeliu9527** [comment]
I also want to know

**Comment by mdouble** [comment]
I also would like to persist my diagrams. I had the idea to store the nodes in a database. However when I do the migration I receive an exception with the following error message: "No suitable constructor was found for entity type 'Rectangle'. The following constructors had parameters that could not be bound to properties of the entity type: cannot bind 'left', 'top', 'right', 'bottom' in 'Rectangle(double left, double top, double right, double bottom)'; cannot bind 'position', 'size' in 'Rectangle(Point position, Size size)'." I am quite puzzled. Anyone any idea? Any help appreciated!

**Reply by mdouble** [reply]
I found the following article that might throw some light on what is going on: https://blog.leem.dev/adventures-in-entity-framework-core-7-constructor-binding.
However I still don't understand why EF has difficulties to bind 'left', 'top', 'right', 'bottom' in 'Rectangle(double left, double top, double right, double bottom). As much as I see the parameter names are the camel-case variant of the Pascal-case property names. So there should be no problem here. 
Regarding the second constructor 'Rectangle(Point position, Size size)' I still have no idea ...

**Reply by mdouble** [reply]
Another update: I tried the workaround suggested in the article and replaced the service with the `CaseInsensitivePropertyParameterBindingFactory.` Unfortunately, this does not change the behavior.

**Reply by mdouble** [reply]
I got around the rectangle problem by explicitly mapping the rectangle properties:
`modelBuilder.Entity<Blazor.Diagrams.Core.Geometry.Rectangle>(b =>
{
    b.Property(e => e.Left);
    b.Property(e => e.Top);
    b.Property(e => e.Right);
    b.Property(e => e.Bottom);  
});
`
This works partially for the `GroupModel` properties, too. However, I have not succeeded mapping the `Children` yet.

**Comment by ecadversario** [comment]
Thanks for the info
